### PR TITLE
fix: update library docs

### LINKED ISSUE #836: Library description does not provide pointers to key terms
Current description is hard to navigate and missing some useful description. We want to improve that to help users to have a chance to self-discover actual use-case for the library

1. Short descriptions for the main concepts like Application Default Credentials. This should provide better value than enumeration of classes right now
2. Missing table of contents, impossible to navigate effectively 
3. Suboptimal structure, where most important artefact description is hidden in the middle of the doc
4. CI links are outdated
5. Still says version is unstable

### PR DESCRIPTION
**Pull request by TimurSadykov**
1. Adding table of contents
2. Adding description of ADC with links instead of class enumeration
3. Switching status to stable
4. remove java7 CI link

Fixes #836  ☕️

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,64 +70,27 @@ libraryDependencies += "com.google.auth" % "google-auth-library-oauth2-http" % "
...
-    AppEngineCredentials.newBuilder()
-        .setScopes(...)
-        .setAppIdentityService(appIdentityService)
-        .build();
-```
+### Application Default Credentials
 
-**Important: `com.google.auth.appengine.AppEngineCredentials` is a separate class from
-`com.google.auth.oauth2.AppEngineCredentials`.**
+This library provides an implementation of [Application Default Credentials](https://cloud.google.com/docs/authentication/getting-started)
```
Comment (lesv) [inline README.md:77]: ADC isn't mentioned on that page.  https://google.aip.dev/auth/4110 explains it at length.

I also asked for repairs to that page.

Comment (lesv) [reply README.md:77]: You might also want to mention https://google.aip.dev/auth

**Review by lesv — APPROVED** [review]
a comment for the readme

File: README.md (hunk 1 above)
Comment (TimurSadykov) [reply README.md:77]: yep, thanks, originally I did not like the link either but forgot about it :)

**Pull request merged**